File: app/src/main/java/com/tasktree/Position.java
package com.tasktree;


import com.google.common.base.Objects;

class Position {
    private int index;

    public Position( int index ) {
        this.index = index;
    }

    public int getIndex( ) {
        return index;
    }

    @Override
    public boolean equals( Object obj ) {
        if (obj == null) return false;
        if (getClass() != obj.getClass()) return false;

        final Position other = (Position) obj;
        return Objects.equal(this.getIndex(), other.getIndex());
    }

    @Override
    public String toString( ) {
        return "Position @index " + index;
    }
}

File: app/src/main/kotlin/com/tasktree/Position.kt
package com.tasktree


import com.google.common.base.Objects

internal open class Position(val index: Int) {

    override fun equals(obj: Any?): Boolean {
        if (obj == null) return false
        if (javaClass != obj.javaClass) return false

        val other = obj as Position?
        return Objects.equal(this.index, other!!.index)
    }

    override fun toString(): String {
        return "Position @index " + index
    }
}
